### ISSUE TITLE
Fix user earns while native Windows and macOS notifications are suppressed

### ISSUE THREAD

**Issue by jonathansampson** [brave/brave-browser#3651]
## Description 
When notifications are disabled for the browser, ad notifications still register as having been displayed and the user is able to earn BAT in spite of never having seen a notification or ad.

## Steps to Reproduce
1. Enable Brave Rewards and Ads
2. Visit _Notifications & Settings_ in Windows Settings
3. Disable notifications for Brave

![image](https://user-images.githubusercontent.com/815158/54096923-f01d4800-437b-11e9-8436-6a63f6b6f691.png)

## Actual result:
The browser believes it is showing ad notifications from time to time.

## Expected result:
The browser ceases to show notifications, or is aware that they aren't visible.

## Reproduces how often: 
Easily.

## Brave version (brave://version info)
0.63.4

### Reproducible on current release:
Ads only enabled in Developer build at the moment.

**Comment by jonathansampson** [comment]
Originally tracked in #2548.

**Comment by tmancey** [comment]
This is possible on iOS/macOS using `getNotificationSettingsWithCompletionHandler:`, for further information see https://developer.apple.com/documentation/usernotifications/unusernotificationcenter/1649524-getnotificationsettingswithcompl?language=objc

**Comment by tmancey** [comment]
Cross Platform Notifications document available at https://docs.google.com/document/d/1JOf26TyIYSZuTNuyc8jDexf7CHyKBCx5ySocaK6fzc4/edit#heading=h.8ihq2tcf777h

**Comment by anthonypkeane** [comment]
I ran into this issue on Mac too
I saw 2 notifications, put on DND and when I turned off DND, the Ads panel said I had received 8 ads

BTW, on Mac, like Slack do, it might be an idea to keep the Ad notification persistent in the notification area so users can see them again.

**Comment by lukemulks** [comment]
We should let the notifications live in the tray, but right now they are not clicking through to Brave from the system tray. 


Brave | 0.68.104 Chromium: 75.0.3770.100 (Official Build) dev (64-bit)
-- | --
Revision | cd0b15c8b6a4e70c44e27f35c37a4029bad3e3b0-refs/branch-heads/3770@{#1033}

**Comment by kjozwiak** [comment]
Quick QA note: Please make sure that `Catalina` is checked as well as per https://github.com/brave/brave-browser/issues/5541 once this lands. When disabling notifications via `Notifications` in `System Preferences`, ads will still be counted as a `view` even though alerts have been blocked.

**Comment by tmancey** [comment]
https://github.com/brave/brave-browser/issues/6786

**Comment by tmancey** [comment]
We have removed the change for macOS which will be fixed as part of another issue

**Comment by LaurenWags** [comment]
removed `OS/macOS` label per discussion with @kjozwiak and https://github.com/brave/brave-browser/issues/3651#issuecomment-551612026

**Comment by GeetaSarvadnya** [comment]
Verification is in progress


Brave | 1.3.99 Chromium: 79.0.3945.130 (Official Build) beta (64-bit)
-- | --
Revision | e22de67c28798d98833a7137c0e22876237fc40a-refs/branch-heads/3945@{#1047}
OS | Windows 10 OS Version 1803 (Build 17134.1006)

- Verified the test plan from https://github.com/brave/brave-core/pull/3084
**Clean profile:**
   - Ensured ad notifications are not shown on windows when `Focus Assist` is enabled
   - Ensured ` Notifications not allowed` text message displayed in ads logs 
      `[4724:17364:0123/232358.660:INFO:ads_impl.cc(1148)] Notification not made: Notifications not 
       allowed`
   - Ensured ad notifications are not shown on Windows when notifications are disabled for the user 
    ```
     [9080:14868:0123/235501.312:INFO:ads_impl.cc(344)] Browser state changed to unidle
     [9080:14868:0123/235501.319:INFO:ads_impl.cc(1148)] Notification not made: Notifications not 
     allowed
    ```
   - Ensured ad notifications are not shown on Windows when notifications are disabled for 
      application
    ```
     [16456:7440:0124/005533.277:INFO:ads_impl.cc(344)] Browser state changed to unidle
     [16456:7440:0124/005533.284:INFO:ads_impl.cc(1148)] Notification not made: Notifications not 
     allowed
    ```

**Comment by GeetaSarvadnya** [comment]
@tmancey Apart from one check rest other checks are validated. Can you explain what is manifest? `Confirm native Ad notifications are not shown on Windows if notifications are disabled by manifest`.
